compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1190,13 +1190,15 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                     // parsing. We try to resolve that ambiguity by attempting resolution in both the
                     // type and value namespaces. If we resolved the path in the value namespace, we
                     // transform it into a generic const argument.
-                    TyKind::Path(qself, path) => {
+                    TyKind::Path(None, path) => {
```

Comment:
Does this change deserve a test?

Reply:
I have added `tests/ui/const-generics/assoc_const_as_type_argument.rs` which has the same behaviour as it does on stable and I think should exercise this codepath.